tests/integration_tests/datasources/test_azure.py

```diff
@@ -45,3 +48,43 @@ def test_azure_eject(session_cloud: IntegrationCloud):
                 session_cloud.cloud_instance.delete_image(snapshot_id)
         else:
             _check_for_eject_errors(instance)
+
+
+@pytest.mark.skipif(PLATFORM != "azure", reason="Test is Azure specific")
+def test_azure_multi_nic_setup(session_cloud: IntegrationCloud) -> None:
+    """Integration test for https://warthogs.atlassian.net/browse/CPC-3999
+
+    Azure should have the primary NIC only route to DNS.
+    Ensure other NICs do not have route to DNS.
+    """
+    us = datetime.datetime.now().strftime("%f")
+    rg_params = AzureParams(f"ci-test-multi-nic-setup-{us}", None)
+    nic_one = AzureCreateParams(f"ci-nic1-test-{us}", rg_params.name, None)
+    nic_two = AzureCreateParams(f"ci-nic2-test-{us}", rg_params.name, None)
+    with session_cloud.launch(
+        launch_kwargs={
+            "image_id": session_cloud.cloud_instance.daily_image(
+                CURRENT_RELEASE.series, image_type=ImageType.GENERIC
+            ),
+        }
+    ) as instance:
+        source = get_validated_source(session_cloud)
+        if source.installs_new_version():
+            instance.install_new_cloud_init(source, clean=True)
+            snapshot_id = instance.snapshot()
+            try:
+                with session_cloud.launch(
```

Comment:
Instead of manually creating a base image, install new cloud-init, create a snapshot and launch the final test image, one can use the pytest fixture [setup_image](https://github.com/canonical/cloud-init/blob/8fe1494493f2cf2f6a04bba633be17937b93af50/tests/integration_tests/test_networking.py#L260). I guess this is a copy-paste from the other test in this file. This could be simplified using the aforementioned setup_image fixture.

Reply:
Updated.